btr-api/src/btr_api/services/auth.py

```diff
@@ -88,3 +88,47 @@ def get_bearer_token(self):
             raise ExternalServiceException(HTTPStatus.SERVICE_UNAVAILABLE,
                                            [{'message': 'Unable to get service account token.',
                                              'reason': err.with_traceback(None)}]) from err
+
+    def get_authorization_header(self, request) -> str:
+        """Gets authorization header from request."""
+        authorization_header = request.headers.get('Authorization', None)
+        if not authorization_header:
+            error = f'Missing authorization header: {request.headers}'
+            self.app.logger.debug('Cannot find authorization_header in request.')
+            raise AuthException(error=error, status_code=HTTPStatus.UNAUTHORIZED)
+
+        return authorization_header
+
+    def is_authorized(self, request, business_identifier: str) -> bool:
+        """Authorize the user for access to the service."""
+        try:
+            auth_token = self.get_authorization_header(request)
+            # make api call
+            headers = {'Authorization': auth_token}
+            auth_url = f'{self.svc_url}/entities/{business_identifier}/authorizations'
+
+            resp = requests.get(url=auth_url, headers=headers, timeout=self.timeout)
+
+            if resp.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                error = f'{resp.status_code} - {str(resp.json())}'
+                self.app.logger.debug('Invalid response from auth-api: %s', error)
+                raise ExternalServiceException(error=error, status_code=HTTPStatus.SERVICE_UNAVAILABLE)
+
+            if resp.status_code != HTTPStatus.OK or not resp.json().get('roles'):
```

Comment:
this should check that roles contains 'edit'. Someone with 'view' only shouldn't have access to the SI form

Reply:
I implemented it like suggested now, but like we discussed, lets follow up on this to see what we need to do to expand allowing roles with just a 'view' to have access.